### PR TITLE
switch to using go1.8

### PR DESCRIPTION
**Pull request by MHBauer**
all I really want is the faster compilation.

### CODE REVIEW

**Comment by duglin** [comment]
Is kube there yet? We should probably match them on this since we vendor in so much of their stuff.

**Comment by MHBauer** [comment]
If it builds and tests, I don't see a reason not to.

**Comment by krancour** [comment]
I think we should delay adoption of 1.8 only if we _know_ there's an issue and not do it simply because k8s may not have done so yet.

Some considerations informing that perspective:

1. Go builds consume dependencies _as source_ and build that source. i.e. We don't link to pre-compiled libraries that might have been compiled with another version of Go. i.e. The possibility of binary incompatibilities is nil.

1. Kubernetes has a _huge_ dependency graph of its own and it's a certainty that _those_ projects upon which we're transitively dependent are built with a wide variety of Go versions. So no matter what version _we're_ on, we're _always_ building with a different version than a substantial amount of code we depend on. If this lack of uniformity actually were any kind of problem, changing what version we use doesn't make that situation any "worse" than it already is.

1. Although I'm not a huge fan of out-of-container builds, we've preserved the ability to do so... and we have no control over what version of Go anyone's system has installed natively. So the "problem" of not being in-sync with whatever version of Go k8s uses to compile upstream is one that some contributors are, for sure, already facing... and if it's not hurting them, there's no reason we can't also use the version _we'd_ like to use in-container.

Bottom line for me is that I see very low likelihood of an issue arising from any lack of uniformity in Go version across projects because that uniformity is already absent and isn't causing problems.

LGTM

**Comment by duglin** [comment]
As long as things pass.... LGTM

**Comment by arschles** [comment]
I am not comfortable saying "as long as things pass, this is ok" or that we are building everything from source, so go 1.8 should be fine. I want to wait until `kubernetes/kubernetes` at least begins using Go 1.8 to compile their tree. Has that happened?

**Comment by MHBauer** [comment]
trying to, tracking in kubernetes/kubernetes#41636

**Comment by MHBauer** [comment]
additionally, the request from go devs to use 1.8 https://github.com/kubernetes/kubernetes/issues/38228

embedded etcd seems to be causing the most issues, and we don't use that in our tests, .... yet.

**Comment by pmorie** [comment]
I would really rather avoid optional problems right now and focus on getting MVP milestones hit.

**Comment by arschles** [comment]
I am marking this in progress, as we are waiting for resolution on https://github.com/kubernetes/kubernetes/issues/38228

**Comment by MHBauer** [comment]
kubernetes/kubernetes#41636 is in, but kubernetes/kubernetes#38228 isn't closed until some tests look clean. Getting close.

**Comment by MHBauer** [comment]
All upstream issues resolved. Builds fine. What does the risk look like?

**Comment by pmorie** [comment]
I'm tied up this week but if we're on 1.8 in upstream I'm all about getting
onto it in this repo.

**Comment by MHBauer** [comment]
Looks like some care is needed when we rebase on 1.7 for developers.
https://github.com/kubernetes/kubernetes/issues/45296

**Comment by krancour** [comment]
> Looks like some care is needed when we rebase on 1.7 for developers.

That isn't impetus to not merge this, is it?
